lifecycle/src/main/scala/it/pagopa/interop/agreementprocess/lifecycle/AttributesRules.scala

```diff
@@ -78,11 +72,14 @@ object AttributesRules {
       consumerAttributes.collect { case a: PersistentVerifiedAttribute => a }
     )
 
-  private def attributesSatisfied(requested: Seq[CatalogAttribute], assigned: Seq[UUID]): Boolean = {
-    requested.forall {
-      case SingleAttribute(value) => assigned.contains(value.id)
-      case GroupAttribute(values) => values.map(_.id).intersect(assigned).nonEmpty
-      case _                      => true
+  private def attributesSatisfied(requested: Seq[Seq[CatalogAttribute]], assigned: Seq[UUID]): Boolean = {
+    requested.forall { list =>
+      {
+        list match {
+          case head :: Nil => assigned.contains(head.id)
+          case values      => values.map(_.id).intersect(assigned).nonEmpty
+        }
+      }
     }
```

Comment:
we don't need to discern the two cases anymore

```suggestion
    requested.forall(_.map(_.id).intersect(assigned).nonEmpty)
```